### PR TITLE
[REM] l10n_in*: removed multi GSTIN

### PR DESCRIPTION
**Pull request by bat-odoo**
In this commit, we have removed the multi-GSTIN feature from the journal.

Multi-unit is created if there is a different branch with the same GSTIN(vat) so
now we decided to create a multi-company instead of this.

because functionally we discuss with Indian CA and we find out that it's good
practice if we create a different company.

The main problem is one company and the same Bank account but we also find a
functional solution using a transfer balance to another company.

so we remove this so our other integrations do not have complexity
like E-invoice, E-waybill, GST E-filing

TaskID - 2723109

### CODE REVIEW

**Review by vinod-singh — APPROVED** [review]

File: addons/l10n_in/models/account_invoice.py [hunk 1]
```diff
@@ -55,16 +55,14 @@ def _post(self, soft=True):
...
-            """Check state is set in company/sub-unit"""
-            company_unit_partner = move.journal_id.l10n_in_gstin_partner_id or move.journal_id.company_id
-            if not company_unit_partner.state_id:
+            if not move.journal_id.company_id.state_id:
                 raise ValidationError(_(
-                    "State is missing from your company/unit %(company_name)s (%(company_id)s).\nFirst set state in your company/unit.",
-                    company_name=company_unit_partner.name,
-                    company_id=company_unit_partner.id
+                    "State is missing from your company %(company_name)s (%(company_id)s).\nFirst set state in your company.",
+                    company_name=move.journal_id.company_id.name,
```
Comment (jco-odoo) [inline addons/l10n_in/models/account_invoice.py:61]:
```suggestion
                    company_name=move.company_id.name,
```

**Review by jco-odoo — COMMENTED** [review]
However more important to get this merged asap already so we clean out...

File: addons/l10n_in/models/account_invoice.py [hunk 2]
```diff
@@ -55,16 +55,14 @@ def _post(self, soft=True):
...
+                    company_id=move.journal_id.company_id.id
                 ))
             elif move.journal_id.type == 'purchase':
-                move.l10n_in_state_id = company_unit_partner.state_id
+                move.l10n_in_state_id = move.journal_id.company_id.state_id
```
[the remark this answers is not in the document]
Comment (jva-odoo) [reply addons/l10n_in/models/account_invoice.py:65]: we have different PR for **place of supply** here https://github.com/odoo/odoo/pull/82011/files

can we merge this PR?

File: addons/l10n_in/models/account_invoice.py [hunk 3]
```diff
@@ -78,9 +76,9 @@ def _post(self, soft=True):
                 move.l10n_in_state_id = self._l10n_in_get_indian_state(move.partner_id)
                 if not move.l10n_in_state_id:
                     move.l10n_in_state_id = self._l10n_in_get_indian_state(move.partner_id)
```
[the remark this answers is not in the document]
Comment (jva-odoo) [reply addons/l10n_in/models/account_invoice.py:78]: it's create when we remove shipping GSTIN here https://github.com/odoo/odoo/pull/81977
@bat-odoo so remove it.

File: addons/l10n_in/models/account_invoice.py [hunk 4]
```diff
@@ -78,9 +76,9 @@ def _post(self, soft=True):
...
-                #still state is not set then assumed that transaction is local like PoS so set state of company unit
+                #still state is not set then assumed that transaction is local like PoS so set state of company
                 if not move.l10n_in_state_id:
-                    move.l10n_in_state_id = company_unit_partner.state_id
+                    move.l10n_in_state_id = move.journal_id.company_id.state_id
```
[the remark this answers is not in the document]
Comment (jva-odoo) [reply addons/l10n_in/models/account_invoice.py:81]: same as above we have different PR for the Place of supply.
we can merge it here OR simply remove this GSTIN

File: addons/l10n_in/models/account_invoice.py [hunk 5]
```diff
@@ -55,16 +55,14 @@ def _post(self, soft=True):
...
-                    "State is missing from your company/unit %(company_name)s (%(company_id)s).\nFirst set state in your company/unit.",
-                    company_name=company_unit_partner.name,
-                    company_id=company_unit_partner.id
+                    "State is missing from your company %(company_name)s (%(company_id)s).\nFirst set state in your company.",
+                    company_name=move.company_id.name,
+                    company_id=move.company_id.id
                 ))
             elif move.journal_id.type == 'purchase':
-                move.l10n_in_state_id = company_unit_partner.state_id
+                move.l10n_in_state_id = move.journal_id.company_id.state_id
```
Comment (jco-odoo) [inline addons/l10n_in/models/account_invoice.py:65]:
```suggestion
                move.l10n_in_state_id = move.company_id.state_id
```

File: addons/l10n_in/models/account_invoice.py [hunk 6]
```diff
@@ -55,16 +55,14 @@ def _post(self, soft=True):
         gst_treatment_name_mapping = {k: v for k, v in
                              self._fields['l10n_in_gst_treatment']._description_selection(self.env)}
         for move in posted.filtered(lambda m: m.country_code == 'IN'):
-            """Check state is set in company/sub-unit"""
-            company_unit_partner = move.journal_id.l10n_in_gstin_partner_id or move.journal_id.company_id
-            if not company_unit_partner.state_id:
+            if not move.journal_id.company_id.state_id:
```
Comment (jco-odoo) [inline addons/l10n_in/models/account_invoice.py:58]:
```suggestion
            if not move.company_id.state_id:
```

**Review by jco-odoo — COMMENTED** [review]
@jva-odoo Would still be good if we can merge this one first.  (before the compute of the state)

**Comment by bat-odoo** [comment]
Hello @jco-odoo 
your last suggested changes are done.
Thanks.

**Comment by bat-odoo** [comment]
Hello @jco-odoo 
Upgrade PR: https://github.com/odoo/upgrade/pull/4107
Both PR is ready can you please check?
Thanks.

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**